scripts/cnv_wdl/somatic/cnv_somatic_allele_fraction_pair_workflow.wdl

```diff
@@ -0,0 +1,262 @@
+    File tumor_bam
+    File tumor_bam_idx
+    File? normal_bam
+    File? normal_bam_idx
+    File tumor_tn_coverage
+    File tumor_called_segments
+    File ref_fasta
+    File ref_fasta_dict
+    File ref_fasta_fai
+    String gatk_jar
+
+    # If WGS, different segment-merging parameters will be used for ACNV
+    Boolean is_wgs
+
+    call GetBayesianHetCoverage {
+        input:
+            common_sites = common_sites,
+            tumor_bam = tumor_bam,
+            tumor_bam_idx = tumor_bam_idx,
+            normal_bam = normal_bam,
+            normal_bam_idx = normal_bam_idx,
+            ref_fasta = ref_fasta,
+            ref_fasta_fai = ref_fasta_fai,
+            ref_fasta_dict = ref_fasta_dict,
+            gatk_jar = gatk_jar
+    }
+
+    call AllelicCNV {
+        input:
+            entity_id = GetBayesianHetCoverage.tumor_entity_id,
+            hets = GetBayesianHetCoverage.tumor_hets,
+            tn_coverage = tumor_tn_coverage,
+            called_segments = tumor_called_segments,
+            is_wgs = is_wgs,
+            gatk_jar = gatk_jar
+    }
+
+    call PlotACNVResults {
+        input:
+            entity_id = GetBayesianHetCoverage.tumor_entity_id,
+            hets = GetBayesianHetCoverage.tumor_hets,
+            tn_coverage = tumor_tn_coverage,
+            acnv_segments = AllelicCNV.acnv_segments,
+            ref_fasta_dict = ref_fasta_dict,
+            gatk_jar = gatk_jar
+    }
+
+    call ConvertACNVResults {
+        input:
+            entity_id = GetBayesianHetCoverage.tumor_entity_id,
+            hets = GetBayesianHetCoverage.tumor_hets,
+            tn_coverage = tumor_tn_coverage,
+            acnv_segments = AllelicCNV.acnv_segments,
+            gatk_jar = gatk_jar
+    }
+
+    output {
+        String tumor_entity_id = GetBayesianHetCoverage.tumor_entity_id
+        File tumor_hets = GetBayesianHetCoverage.tumor_hets
+        File acnv_segments = AllelicCNV.acnv_segments
+    }
+}
+
+# Call heterozygous SNPs in the normal and then collect allele counts in the tumor for each called position
+# Entity IDs can be the same value and are derived from the BAMs if not provided
+# If the normal BAM is not provided, then tumor-only mode is run
+task GetBayesianHetCoverage {
+    File common_sites
+    File tumor_bam
+    File tumor_bam_idx
+    File? normal_bam
+    File? normal_bam_idx
+    Float? stringency
+    Int? read_depth_threshold
+    File ref_fasta
+    File ref_fasta_dict
+    File ref_fasta_fai
+    String gatk_jar
+    Int? mem
+
+    # Sample names are derived from the bam filenames
+    String tumor_base_filename = sub(sub(sub(tumor_bam, "gs://", ""), "[/]*.*/", ""), "\\.bam$", "")
+    String normal_base_filename = sub(sub(sub(select_first([normal_bam, ""]), "gs://", ""), "[/]*.*/", ""), "\\.bam$", "") # Evaluates to "" if tumor-only and is not used
+
+    # TODO the below if-else block can be cleaned up once Cromwell support for optionals/conditionals improves
```

Comment:
Are you sure this works?  My understanding is that you actually need to check if the value is ``null`` not ``""``